cloudimg/aws.py

```diff
@@ -647,7 +668,7 @@ def register_image(self, snapshot, metadata):
             EnaSupport=metadata.ena_support,
             SriovNetSupport=metadata.sriov_net_support,
             BillingProducts=metadata.billing_products,
-            BootMode=metadata.boot_mode
+            BootMode=metadata.boot_mode.value
```

Comment:
I tried to check documentation and it is not clear to me what will happen in the situation
`BootMode=None`
* https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2/client/register_image.html#EC2.Client.register_image
@JAVGan @jajreidy Can you check with dev account?

Reply:
```
For AMIs with no boot mode parameter, the instances launched from these AMIs use the default boot mode value of the instance type.
```

Ref: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ami-boot.html

So I believe it depends on the EC2 instance type. Each one may have a different "default"

Reply:
@lslebodn you're correct in the assumption we can't pass the `None` value to `BootMode`.

It fails as shown below:

```
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter BootMode, value: None, type: <class 'NoneType'>, valid types: <class 'str'>
```

I've patched it [here](https://github.com/release-engineering/cloudimg/compare/c494d061f43f5e8ee8c41b1ba3c7e80f34ea0cfb..7132a0955c66c571dad38eb276231216c8bc4d71), PTAL again

It's now passing the manual test.

I've added a new automated test for the `None` case as well